### PR TITLE
Jw vector statistical phasing

### PR DESCRIPTION
**Pull request by gbggrant**
First version of Vector Statistical Phasing pipeline.
Contains a WDL to do the statistical phasing itself and an uber WDL to wrap read-backed and statistical phasing.

### CODE REVIEW

**Review by jessicaway — APPROVED** [review]
A few minor comments, but this looks great!

File: dockerfiles/README.md [hunk 1]
```diff
@@ -52,5 +55,6 @@ A farm5-specific docker image, providing dependencies for the iRODS import and b
 - samplevcftozarr - `docker build . --tag=samplevcftozarr:1.0`
 - sampleselectvariants - `docker build . --tag=sampleselectvarians:1.0`
 - whatshap - `docker build . --tag=whatshap:1.0`
+- shapeit4 - `docker build . --tag=shapeit4:4.1.3`
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply dockerfiles/README.md:62]: Yup, good catch.

File: pipelines/phasing-vector/gcp/Phasing.wdl [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          called_sites_zarr = called_sites_zarr,
+          phased_sites_zarr = phased_sites_zarr,
+          contig = chromosome,
+          reference = reference,
+          runTimeSettings = runTimeSettings
+      }
+    }
+
+    # combine samples
+    Array[File] phased_sample_vcfs = ReadBackedPhasing.phased_sample_vcf
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply pipelines/phasing-vector/gcp/Phasing.wdl:65]: Another good catch.

File: pipelines/phasing-vector/gcp/ReadBackedPhasing.wdl [hunk 3]
```diff
@@ -2,66 +2,105 @@ version 1.0
...
 import "../../../structs/gcp/RunTimeSettings.wdl"
 import "../../../structs/ReferenceSequence.wdl"
-import "../../../tasks/gcp/ReadBackedPhasingTasks.wdl" as Tasks
+import "../../../tasks/gcp/Tasks.wdl" as Tasks
+import "../../../tasks/gcp/ShortReadAlignmentTasks.wdl" as ShortReadAlignmentTasks
+import "../../../tasks/gcp/SNPGenotypingTasks.wdl" as SNPGenotypingTasks
+import "../../../tasks/gcp/ReadBackedPhasingTasks.wdl" as ReadBackedPhasingTasks
 
 workflow ReadBackedPhasing {
   String pipeline_version = "0.0.0"
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply pipelines/phasing-vector/gcp/ReadBackedPhasing.wdl:17]: yup

File: dockerfiles/CohortVcfToZarr/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.7.2
+
+RUN pip install --upgrade pip
+
+COPY requirements.txt .
+RUN pip3 install numpy==1.19.1
+RUN pip3 install zarr==2.4.0
+RUN pip3 install scikit-allel==1.3.1
```
Comment (alimanfoo) [inline dockerfiles/CohortVcfToZarr/Dockerfile:8]: Small suggestion but if you are copying the requirements file into the docker image then you could just do this.
```suggestion
COPY requirements.txt .
RUN pip3 install -r requirements.txt
```

**Review by alimanfoo — APPROVED** [review]
Thanks @gbggrant, looks good as far as I can tell. One small suggestion. Also wondered about the Python scripts, these are ending up in two places within this repo, one within the scripts directory and one within the dockerfiles, didn't know if there was a way to pull them into the docker image from the scripts folder. No problem if that's awkward.

File: dockerfiles/CohortVcfToZarr/Dockerfile (hunk 4 above)
Comment (gbggrant) [reply dockerfiles/CohortVcfToZarr/Dockerfile:8]: I like it!

**Comment by gbggrant** [comment]
@alimanfoo yes, the reason that there are duplicates of your scripts in the docker directories has to do with the way docker builds.  You cannot access a directory outside of the build context (for instance with a relative path).  I think the easiest/best thing is just to move the files from the scripts into the docker-appropriate directory.  I may just make a separate PR for that though.

**Pull request merged**